artemis/src/main/java/tech/pegasys/artemis/cli/util/YamlConfigFileDefaultProvider.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.cli.util;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.scanner.ScannerException;
+import picocli.CommandLine;
+import picocli.CommandLine.IDefaultValueProvider;
+import picocli.CommandLine.Model.ArgSpec;
+import picocli.CommandLine.Model.CommandSpec;
+import picocli.CommandLine.Model.OptionSpec;
+import picocli.CommandLine.ParameterException;
+
+public class YamlConfigFileDefaultProvider implements IDefaultValueProvider {
+
+  private static final String FILE_NOT_FOUND_ERROR_MSG =
+      "Unable to read yaml configuration. File not found: ";
+  private static final String IO_ERROR_MSG_FMT =
+      "Unexpected IO error while reading yaml configuration file [%s]: %s";
+  private static final String INVALID_YAML_MSG_FMT =
+      "Unable to read yaml configuration. Invalid yaml file [%s]: %s";
+  private static final String UNKNOWN_OPTIONS_YAML_FMT =
+      "Unknown %s in yaml configuration file: %s";
```

Comment:
nit: It would be better to inline these - they're only used once and the number of `%s` placeholders needs to match up with the number of parameters so maintaining locality is really useful.

Reply:
👍 Inlining.